### ISSUE TITLE
v1.11 Endgame

### ISSUE THREAD

**Issue by artursouza** [dapr/dapr#6301]
## Release team
Release lead: @artursouza
Release lead shadow(s): @nyemade-uversky @robertojrojas
Perf test lead: @JoshVanL @robertojrojas
Long haul test: @halspang
Build manager: @JoshVanL

Read about the [Dapr release lifecycle](https://github.com/dapr/community/blob/master/release-process.md)

## Milestone dates
Code freeze: ~05/01/2023~ - 05/18/2023
Endgame starts: 05/19/2023
Endgame ends: 06/05/2023
1.11.0 release: 06/06/2023

Release Tasks:

------------------
ETA for all below: TBD

- [x] All P0 issues below are resolved PRIOR to code freeze:
  - [x] Actor reminder & timer serialization regressions are fixed in 1.10.6 and merged into master - @artursouza
  - [x] ~gRPC service invocation error message regression (detected by Java SDK) is fixed in 1.10.6 and merged into master~ Not a regression, it is due to a legit hotfix and error message is not how it was in 1.9 - @artursouza
  - [x] Perf regressions in master branch - @yaron & @JoshVanL 
  - [ ] ~Flakiness of placement unit tests https://github.com/dapr/dapr/issues/6265~ - @JoshVanL 
  - [ ] ~Flakiness of external service invocation E2E tests on Windows~ - @yaron2
  - [x] Eventgrid component regression on metadata change - @berndverst 
  - [x] Bulk secret regression detected by Java SDK - @mukundansundar
  - [x] Add stable_components build flavor tag for newly stable components - @berndverst 
  - [x] ~https://github.com/dapr/dapr/issues/6316~ https://github.com/dapr/dapr/pull/6325 - @artursouza 
  - [x] https://github.com/dapr/dapr/pull/6342 - @ItalyPaleAle @berndverst 
  - [x] https://github.com/dapr/dapr/issues/6324 - @JoshVanL 
- [x] All MUST HAVE feature and improvements planned for 1.11 are merged:
  - [x] https://github.com/dapr/java-sdk/pull/846 - @artursouza @mukundansundar 
  - [x] https://github.com/dapr/dapr/pull/6280
  - [x] https://github.com/dapr/dapr/pull/6225
  - [x] https://github.com/dapr/dapr/issues/6099 - @ItalyPaleAle
- [x] CLI changes blocked by runtime RC
  - [x] https://github.com/dapr/cli/pull/1276
  - [x] CLI changes for external service invocation: add new CRD to install & upgrade commands - @sicoyle (pending runtime RC)
- [x] MUST HAVE for SDK releases:
  - [x] https://github.com/dapr/dotnet-sdk/pull/1077
  - [x] https://github.com/dapr/dotnet-sdk/pull/1080
  - [x] https://github.com/dapr/go-sdk/pull/391
  - [x] https://github.com/dapr/go-sdk/pull/387
- [x] NICE TO HAVES for runtime:
  - [x] Add Kitex binding component
  - [ ] ~Add Nacos binding component~ - needs to be tested first, moved to 1.12
  - [ ] ~Add back Temporal~ - we cannot add Temporal because it has not been tested with latest Workflow changes
- [x] Code freeze at 05/18/2023
- [x] Check all new components are registered in dapr/dapr @artursouza 
- [x]  E2E, certification and perf tests are passing @artursouza
  - [x] E2E: https://github.com/dapr/dapr/actions/runs/5020273373
  - [x] Perf: https://github.com/dapr/dapr/actions/runs/5157136767
- [x] Update components-contrib certification tests go.mods with dapr/dapr 1.11.0-rc.1 commit hash  @berndverst  
- [x] Create release-1.11 branch and cut RC 1.11.0-rc.1 for component-contrib 
- [x] Update dapr/component-contrib pkg to 1.11.0-rc.1 in dapr/dapr master branch 
- [x] Create release-1.11 branch for Dapr runtime 
- [x] Verify helm chart and cut the 1.11.0-rc.1 release for Dapr runtime 
- [x] Create release-1.11 branch for CLI 
- [x] Update dapr/dapr pkg to 1.11.0-rc.1 in dapr/cli release-1.11 branch 
- [x] Cut the 1.11.0-rc.1 release for CLI
- [x] Trigger GitHub actions in dapr/homebrew-tap (GitHub disables Actions after 1mo without commits) - @artursouza 
- [x] Notify users about dapr/dapr 1.11.0-rc.1 via Discord @artursouza 
- [x] Validate the upgrade path from the previous release -- CRDs updated, no deletion of components/configs etc. (~This can partially be accomplished by updating the CLI upgrade_test.go~) @pravinpushkar 
- [ ] ~Validate dapr init in self hosted mode for RC from GHCR and default registry~ @DeepanshuA
- [ ] ~Validate dapr init in k8s mode for RC from GHCR and default registry~ @DeepanshuA 
- [ ] ~Validate renew-certificate command with a cluster with a Dapr enabled app~ @yaron2
- [x] Create RC for installer-bundle @shivamkm07 
- [x] Validate RC for installer bundle @shivamkm07 
- [x] Update proto and cut the rc release for Java SDK  @artursouza 
- [x] Update proto ~and cut the rc release~ for Python SDK  @berndverst 
- [x] Update proto ~and cut the rc release~ for Go SDK  @JoshVanL 
- [x] Update proto and cut the rc release for dotnet SDK  @halspang 
- [x] Update proto ~and cut the rc release for~ cpp SDK  @pravinpushkar 
- [x] Update proto and cut the rc release for rust SDK @JoshVanL 
- [X] Cut the rc release for Dashboard @artursouza
- [x] Update longhaul tests with RC for runtime and dotnet SDK @halspang 
- [x] Update quickstarts/tutorials automated validation and check it works - covers Linux Standalone/ K8s / Darwin MacOS.  @yaron2 @paulyuk 
- [x] Test and validate tutorials in Linux ARM64 k8s and self-hosted - @JoshVanL 
- [ ] Test and validate tutorials in Windows - @artursouza  
- [x] Update Configuration API quickstart - @shivamkm07 
- [x] Create Cryptography API quickstart - @ItalyPaleAle
- [x] Create Python quickstart for Workflows - @DeepanshuA  
- [x] Validate Dapr CLI non-slim scenarios on Windows @shubham1172 
- [x] Validate longhaul metrics @tmacam  
- [x] Test and validate the .NET SDK - run through samples @halspang 
- [x] Test and validate the Java SDK - run through samples @artursouza
- [x] Test and validate the Python SDK - run through samples  @berndverst
- [x] Test and validate the Go SDK - run through samples @robertojrojas  
- [x] Test and validate Dashboard, local mode and kubernetes hosted @artursouza 
- [x] Update certification tests in components-contrib release branch (prior to merge back into master) to use the RC and Go SDK release. @berndverst 
- [x] Confirm with maintainers that each repo has milestone set for all relevant issues & PRs, then generate release notes
  - [x] CLI - @mukundansundar 
  - [x] runtime - @artursouza @yaron2 @mukundansundar 
  - [x] components-contrib - @artursouza @yaron2 @berndverst @DeepanshuA 
  - [x] .NET SDK - @halspang 
  - [X] Java SDK - @artursouza 
  - [ ] ~JS SDK~ - @shubham1172 
  - [X] Python SDK - @berndverst 
  - [x] Go SDK - @yaron2 
  - [x] Check all milestones for each repo above are up to date before cutting the release notes. @artursouza
- [ ] Validate FOSSA on each repo (contrib is included in runtime)
  - [x] CLI - @mukundansundar 
  - [x] runtime - @artursouza @yaron2 @mukundansundar 
  - [x] .NET SDK - @halspang 
  - [x] Java SDK - @artursouza 
  - [ ] ~JS SDK~ - @shubham1172 
  - [x] Python SDK - @berndverst 
  - [x] Go SDK - @yaron2 
- [x] Edit and complete release-notes on HackMD (everyone)
- [x] Verify 1.11 documentation @msfussell
ETA for all above: ~06/05/2023~ 06/12/2023

------------------
RELEASE DAY: ~06/05/2023~ 06/12/2023
- [x] Release Dashboard @artursouza, update dapr/dapr's helm chart to remove RC suffix
- [x] Validate breaking changes section in release notes @artursouza
- [x] Merge release notes into release branch @artursouza 
- [x] Release Dapr 1.11 @artursouza 
- [x] Release CLI 1.11 @mukundansundar @yaron2 
- [ ] Release Installer Bundle 1.11 @shivamkm07 
- [x] Create blog post @nyemade-uversky 
- [x] Create release-1.11 branch for quickstarts, update version tag in the images, create v1.11.0 tag @paulyuk 
- [x] Release cpp-sdk @pravinpushkar @artursouza
- [x] Release js-sdk @shubham1172 
- [x] Release go-sdk @yaron2 
- [x] Release java-sdk @artursouza 
- [x] Release python-sdk @berndverst 
- [x] Release rust-sdk @artursouza as per approval by STC
- [x] Release dotnet-sdk and release nuget packages  @artursouza as per approval by STC
- [x] Create components-contrib release from Tag. Copy components release notes section into the description of the release. @ItalyPaleAle  
- [x] Validate DevContainers in following repos
  - [x] dapr/dapr @ItalyPaleAle 
  - [x] dapr/components-contrib @ItalyPaleAle 
  - [x] dapr/cli @ItalyPaleAle 
- [x] Update certification tests in components-contrib release branch (prior to merge back into master) to use latest runtime and Go SDK release. @ItalyPaleAle  
- [x] Generate new Java docs based on the release version of the Java SDK @artursouza

------------------
ANNOUNCEMENT DAY: 06/06/2023
- [x] Update supported versions table in docs @greenie-msft 
- [x] Merge this quickstarts PR https://github.com/dapr/quickstarts/pull/778
- [x] Release blog @yaron2  
- [x] Tweet release announcement @marcduiker  
- [x] Discord release announcement @marcduiker  
- [x] Update versions in the E2E tests in CLI to the latest runtime and dashboard versions @shubham1172  
- [ ] Merge (NOT SQUASH MERGE/AUTO MERGE) back release-1.11 branch into master for required repos @ItalyPaleAle @artursouza 
- [x] Celebrate (everyone)

**Comment by mukundansundar** [comment]
Release blocker https://github.com/dapr/cli/issues/1289

**Comment by JoshVanL** [comment]
>  Test and validate tutorials in Linux ARM64 k8s and self-hosted - @JoshVanL

Ran through, and they look good :+1: Small nit here https://github.com/dapr/docs/pull/3459

**Comment by pravinpushkar** [comment]
@artursouza 
> Validate the upgrade path from the previous release -- CRDs updated, no deletion of components/configs etc.

Validated, all good here.

**Comment by artursouza** [comment]
Java SDK FOSSA is clear, no action was required: https://app.fossa.com/reports/e4f9abb1-8641-4940-8e16-f46e097cf701
